design/ErrorHandling.md

```diff
@@ -23,59 +23,60 @@ There are several types of WebGPU calls that get their errors handled differentl
-The application "eventually" recieves these errors, asynchronously via a callback.
-
-These errors are all delivered through one stream - not directly associated with the objects or operations that produced them.
-They should not be used by applications to recover from expected, recoverable errors.
-Instead, the error log may be used for handling unexpected errors in deployment, for bug reporting and telemetry.
-
-```
-enum GPULogEntryType {
-    "device-lost",
-    "validation-error",
-    "recoverable-out-of-memory",
+## *Telemetry*: Error Logging
+
+```webidl
+[
+    Constructor(DOMString type, GPUValidationErrorEventInit gpuValidationErrorEventInitDict),
+    Exposed=Window
+]
+interface GPUValidationErrorEvent : Event {
+    readonly attribute DOMString message;
 };
 
-interface GPULogEntry {
-    readonly attribute GPULogEntryType type;
-    readonly attribute any object;
-    readonly attribute DOMString? reason;
+dictionary GPUValidationErrorEventInit : EventInit {
+    required DOMString message;
 };
 
-callback GPULogCallback = void (GPULogEntry error);
-
-partial interface GPUDevice {
-    attribute GPULogCallback onLog;
+partial interface GPUDevice : EventTarget {
+    attribute EventHandler onvalidationerror;
 };
 ```
 
-`GPUObjectStatusType` makes a distinction between several error types:
+`WebGPUDevice` `"validationerror"` -> `GPUValidationErrorEvent`:
+Fires when a non-fatal error occurs in the API, i.e. a validation error (including if an "invalid" object is used).
+
+When there is a validation error in the API (including operations on "invalid" WebGPU objects), an error is logged.
+When a validation error is discovered by the WebGPU implementation, it may fire a `"validationerror"` event on the `GPUDevice`.
+These events should not be used by applications to recover from expected, recoverable errors.
+Instead, the error log may be used for handling unexpected errors in deployment, for bug reporting and telemetry.
+
+`"validationerror"` events are all delivered through the device.
+They are not directly associated with the objects or operations that produced them.
 
- - `"validation-error"`: validation of some WebGPU call failed - including if an argument was an "invalid object".
-   (Either the application did something wrong, or it chose not to recover from a recoverable error.)
- - `"device-lost"`: the `GPUDevice` cannot be used anymore.
-   This may happen if the device is destroyed by the application, reclaimed by the browser, or something goes terribly wrong.
-   (An application may request a new device, or choose to fallback to other content.)
- - `"out-of-memory"`: an allocation failed because too much memory was used by the application (CPU or GPU).
-   This includes recoverable out of memory errors that aren't opt-ed in to be handled by the application when the resource was created.
+The `"validationerror"` event always fires on the main thread (Window) event loop.
```

Comment:
You may want to use the existing WebGL task source as defined in https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.15 and says something like

> The task source for all tasks queued [HTML] in this section is the WebGL task source.

For info, in Picture-in-Picture, I wrote:

> The task source for all the tasks queued in this specification is the media element event task source of the video element in question.

For more, check out https://cs.chromium.org/chromium/src/third_party/blink/public/platform/task_type.h